lib/elixir/pages/Guards.md

```diff
@@ -82,7 +83,7 @@ not_nil_head?(["some_value", "another_value"])
 #=> false
 ```
 
-Even though the head of the list is not `nil`, the first clause for `not_nil_head?/1` fails, triggering the second clause which returns `false`. To make the guard behave correctly, you must ensure that the guard evaluates to `true`, like so:
+Even though the head of the list is not `nil`, the first clause for `not_nil_head?/1` fails because the expression is evaluated to `"some_value"` and not to `true` as guards are expected to return, therefore triggering the second clause which returns `false`. To make the guard behave correctly, you must ensure that the guard evaluates to `true`, like so:
```

Comment:
The new sentence is too long.

```suggestion
Even though the head of the list is not `nil`, the first clause for `not_nil_head?/1` fails because the expression does not evaluate to `true`, but to `"some_value"`, therefore triggering the second clause which returns `false`. To make the guard behave correctly, you must ensure that the guard evaluates to `true`, like so:
```

Reply:
done